contracts/implementations/nodes/AftermarketDevice.sol

```diff
@@ -157,52 +171,136 @@ contract AftermarketDevice is
             );
         }
 
-        // Validate request and transfer funds to foundation
-        // This transfer is at the end of the function to prevent reentrancy
-        _validateMintRequest(msg.sender, devicesAmount);
+        // Validate license and transfer funds to foundation
+        _validateMintRequest(msg.sender, msg.sender, devicesAmount);
     }
 
-    /// @notice Claims the ownership of a list of aftermarket devices to a list of owners
-    /// @dev Caller must have the admin role
-    /// @dev This contract must be approved to spend the tokens in advance
-    /// @param adOwnerPair List of pairs AD-owner
-    function claimAftermarketDeviceBatch(
-        AftermarketDeviceOwnerPair[] calldata adOwnerPair
-    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    /**
+     * @notice Mints aftermarket devices in batch by an authorized address
+     * The manufacturer node owner must grant the minter privilege to the caller
+     * @param manufacturerNode Parent manufacturer node id
+     * @param adInfos List of attribute-info pairs and addresses associated with the AD to be added
+     *  addr -> AD address
+     *  attrInfoPairs[] / attribute
+     *                  \ info
+     */
+    function mintAftermarketDeviceAuthorized(
```

Comment:
Yeah, gas. And also the modifier `onlyRole(MANUFACTURER_ROLE)`, but I think this is kinda redundant since we verifiy the license that is way important.

But on second though, I could only use `hasPrivilege` bc it returns true if the address is the owner of the token. I think I would take simplicity over gas saving this time.

Reply:
Just realized I have to fix the logic in `hasPrivilege`

Reply:
Uh oh.